### PR TITLE
Add a Julia interface for MC21

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

**Comment by amontoison** [comment]
MC21 is a relevant routine if we want to compute a ILU0 preconditioner for instance.
I use the version provided by CUDA (`zfd`) as example in the documentation of Krylov.jl.

File: src/hsl_mc21.jl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+"""
+function mc21(A::SparseMatrixCSC; sym::Bool=false)
+  n, m = size(A)
+  n == m || error("A is not square!")
+  if sym
+    B = A
+  else
+    B = similar(A)
+    transpose!(B, A)
+  end
```
Comment (dpo) [inline src/hsl_mc21.jl:33]: That's quite costly! Can we say instead that we find a column permutation and work only with A?

Comment (amontoison) [reply src/hsl_mc21.jl:33]: Yes, good idea!
The algorithm was developed for the CSR format  and not CSC.

**Comment by dpo** [comment]
@JSOBot runtests

**Comment by dpo** [comment]
@JSOBot runtests

**Comment by dpo** [comment]
MC21 ran this time.

**Pull request merged**